gateways/istio-egress/templates/_affinity.tpl

```diff
@@ -1,28 +1,35 @@
 {{/* affinity - https://kubernetes.io/docs/concepts/configuration/assign-pod-node/ */}}
 
-{{- define "nodeaffinity" }}
+{{- define "gatewaynodeaffinity" }}
   nodeAffinity:
     requiredDuringSchedulingIgnoredDuringExecution:
-    {{- include "nodeAffinityRequiredDuringScheduling" . }}
+    {{- include "gatewayNodeAffinityRequiredDuringScheduling" . }}
     preferredDuringSchedulingIgnoredDuringExecution:
-    {{- include "nodeAffinityPreferredDuringScheduling" . }}
+    {{- include "gatewayNodeAffinityPreferredDuringScheduling" . }}
 {{- end }}
 
-{{- define "nodeAffinityRequiredDuringScheduling" }}
+{{- define "gatewayNodeAffinityRequiredDuringScheduling" }}
       nodeSelectorTerms:
       - matchExpressions:
         - key: beta.kubernetes.io/arch
           operator: In
           values:
-        {{- range $key, $val := .Values.global.arch }}
+        {{- range $key, $val := .root.Values.global.arch }}
```

Comment:
what is .root ?

Reply:
If you look later down when we call it: `      {{- include "gatewaynodeaffinity" (dict "root" $ "nodeSelector" $gateway.nodeSelector) | indent 6 }}`

We pass a `.root`

This template is so complicated I was worried changing it would break it -- probably not a good state to be in. I can try to clean this up.

Ideally i think we would have no affinity stuff, and just make users kustomize it